garden-backend-service/src/api/routes/users.py

```diff
@@ -62,71 +63,124 @@ async def update_user_info(
     return user_response
 
 
-@router.get("/gardens/saved")
-async def get_saved_garden_dois(
-    authed_user: User = Depends(authed_user),
+@router.get("/{user_uuid}/saved/gardens", response_model=list[GardenMetadataResponse])
+async def get_saved_gardens(
+    user_uuid: UUID,
     db: AsyncSession = Depends(get_db_session),
-) -> list[str]:
-    return await _get_saved_garden_dois(authed_user, db)
+) -> list[GardenMetadataResponse]:
+    """Fetch the users list of saved gardens."""
+    user: User | None = await User.get(db, identity_id=user_uuid)
 
+    if user is None:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail=f"No user with uuid {user_uuid} found.",
+        )
 
-@router.patch("/gardens/saved/{doi:path}")
+    # We need to explicitly load the saved_gardens field
+    await db.refresh(user, ["saved_gardens"])
+    return [GardenMetadataResponse.from_orm(garden) for garden in user.saved_gardens]
+
+
+@router.put(
+    "/{user_uuid}/saved/gardens/{doi:path}", response_model=list[GardenMetadataResponse]
+)
 async def save_garden(
+    user_uuid: UUID,
     doi: str,
     authed_user: User = Depends(authed_user),
     db: AsyncSession = Depends(get_db_session),
-) -> list[str]:
-    user: User = await User.get(db, identity_id=authed_user.identity_id)
+) -> list[GardenMetadataResponse]:
+    """Add a garden to the user's list of saved gardens by doi."""
+    user: User | None = await User.get(db, identity_id=user_uuid)
     garden: Garden | None = await Garden.get(db, doi=doi)
 
+    if user_uuid != authed_user.identity_id:
+        raise HTTPException(
+            status_code=status.HTTP_401_UNAUTHORIZED,
+            detail=f"Not authorized to save a garden for user {user_uuid}.",
+        )
+
+    if user is None:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail=f"No user with user_uuid {user_uuid} found.",
+        )
+
     if garden is None:
         raise HTTPException(
             status_code=status.HTTP_404_NOT_FOUND,
             detail=f"No garden with doi {doi} found.",
         )
 
-    stmt = insert(users_saved_gardens).values(user_id=user.id, garden_id=garden.id)
+    await db.refresh(user, ["saved_gardens"])
+    for saved_garden in user.saved_gardens:
```

Comment:
I don't mind either way (up to you whichever syntax you think is clearer) but in case you're interested, I believe that we could take advantage of our `AsyncAttrs` mixin here like 
```python
for saved_garden in await user.awaitable_attrs.saved_gardens: 
    ...
```

Reply:
that being said, I don't think we actually want this loop, because we want PUTs to be idempotent; saving a garden you've already saved should be a no-op. Something like this probably "just works" with the orm, though I haven't tested it 
```python
await db.refresh(user, ["saved_gardens"])
if garden not in user.saved_gardens:
    user.saved_gardens.append(garden) 
```